src/back/routes/project.ts

```diff
@@ -129,3 +136,41 @@ async function deleteLink(req: WithAuth<Request<{ link_id: string }>>): Promise<
 
   return await ProjectService.deleteLink(link_id)
 }
+async function addMilestone(req: WithAuth): Promise<ProjectMilestone> {
+  const user = req.auth!
+  const { milestone } = req.body
+  validateId(milestone.project_id)
+  const project = await ProjectService.getProject(milestone.project_id)
+  if (!project) {
+    throw new RequestError(`Project "${milestone.project_id}" not found`, RequestError.NotFound)
+  }
+
+  const formattedMilestone = {
+    ...milestone,
+    delivery_date: new Date(milestone.delivery_date),
+  }
+  const parsedMilestone = ProjectMilestoneInCreationSchema.safeParse(formattedMilestone)
+  if (!parsedMilestone.success) {
+    throw new RequestError(`Invalid milestone: ${parsedMilestone.error.message}`, RequestError.BadRequest)
+  }
+
+  return await ProjectService.addMilestone(parsedMilestone.data, user)
+}
+
+async function deleteMilestone(req: WithAuth<Request<{ milestone_id: string }>>): Promise<string | null> {
+  const user = req.auth!
+  const milestone_id = req.params.milestone_id
+  validateId(milestone_id)
+  const milestone = await ProjectMilestoneModel.findOne<ProjectMilestone>(milestone_id)
+  if (!milestone) {
+    throw new RequestError(`Milestone "${milestone_id}" not found`, RequestError.NotFound)
+  }
+  const project = await ProjectService.getProject(milestone.project_id)
```

Comment:
I think you can avoid checking for the project since you use `validateCanEditProject`, and also if the milestone exists, then the `project_id` must exist cause its a fk